### PR TITLE
[RW-8585][risk=low] Reduce request deadline to Terra, except for create/clone workspace

### PR DESCRIPTION
**Pull request by calbach**
Allowing a request to stall out for longer than the GAE request deadline causes a lot of issues in the logs and can obfuscate the cause. Sometimes this will also seemingly result in the failure of other requests on the same task. Setting a tighter deadline should hopefully cause doomed requests to fail earlier, and with a more targeted stack trace / error message.

This PR introduces a "lenient" API client and workspaces API, specifically for create/clone. This is the idea originally suggested on https://github.com/all-of-us/workbench/pull/6409.

Note: 40s is still **very**  high. Ideally we would get this down to 20 seconds or lower, but that's probably too big of a jump for now.

Tested locally be creating a new workspace

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java [hunk 1]
```diff
@@ -117,10 +117,13 @@ public static class FireCloudConfig {
     // This value specifies the information we hand to Terra as our AppId header.
     // It is primarily used for metrics gathering information.
     public String xAppIdValue;
-    // The name of the VPC service perimeter to create our Terra GCP projects inside,
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:120]: heh thanks

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java [hunk 2]
```diff
@@ -142,11 +142,11 @@ public FireCloudServiceImpl(
     this.groupsApiProvider = groupsApiProvider;
     this.nihApiProvider = nihApiProvider;
     this.endUserWorkspacesApiProvider = endUserWorkspacesApiProvider;
+    this.endUserLenientTimeoutWorkspacesApiProvider = endUserLenientTimeoutWorkspacesApiProvider;
     this.serviceAccountWorkspaceApiProvider = serviceAccountWorkspaceApiProvider;
     this.statusApiProvider = statusApiProvider;
     this.termsOfServiceApiProvider = termsOfServiceApiProvider;
     this.endUserStaticNotebooksApiProvider = endUserStaticNotebooksApiProvider;
-    this.serviceAccountStaticNotebooksApiProvider = serviceAccountStaticNotebooksApiProvider;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java:149]: was this unused?  Should delete `SERVICE_ACCOUNT_STATIC_NOTEBOOKS_API` and all refs to it as well.

**Review by jmthibault79 — APPROVED** [review]
I like this approach

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java (hunk 2 above)
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java:149]: Yep, unused. Good call - done.

**Pull request merged**